src/visualiser/svg.js

```diff
@@ -0,0 +1,192 @@
+     * @param {number} LSvalue the LearningStyleValue form which to calculate the x value of the circle
+     * @returns {number} Returns the x coordinate of the circle
+     */
+    function circleXValue(LSvalue){
+        let xPos = svgLineSpace; // add the ofset from the left of the svg (the lines start some length inside the grey) this is equal to the -11 position
+        let percent = (LSvalue + 11) / RANGEWIDTH; // from -11 to 11, how many % is the position into the line? 0%=-11, 100%=11
+        let lineLength = svgWidth - (2 * svgLineSpace); // the length of the line
+        xPos += lineLength * percent; // how far into the line must the center of the circle be placed
+        return xPos;
+    }
+
+    /**
+     * @summary finds how many elements are close to each other and increases their size accordingly
+     * @param {Array} arrCircleSize the input array, this contains
+     * @returns {Array} Returns an array with the radius(size) of the circles to be made, so alle the circles can be seen
+     */
+    function closeby(arrCircleSize){ // something is wrong in this function, but it still kinda works
+        let A = new Array();
+        for (let i = 0; i < arrCircleSize.length; i++) {
+            A[i] = 0;
+            for (let j = i; j < arrCircleSize.length; j++) {
+                if (i !== j){
+                    if (range(arrCircleSize[i], arrCircleSize[j])){
+                        A[i]++;
+                        A[j]++;
+                    }
+                }
+            }
+            A[i]++; // add sig selv, så kan vi gange med det, ellers ganger vi med 0 og så kommer der problemer
+        }
+        // calc the circlesize in pixels by the size e.g. 1, 2, 3... depends on how many values are close to eachother
+        A.forEach((size, idx, arr) => {
+            arr[idx] = 0.25 * svgLineSpace * Math.pow(size, 0.6);
+        });
+        return A;
+    }
+    /**
+     * @summary Returns boolean whether the distance between a & b is smaller than 'afstand'
+     * @param {number} a the first number
+     * @param {number} b the second number
+     * @returns {boolean} Returns boolean whether the distance between a & b is smaller than 'afstand'
+     */
+    function range(a, b){
+        let afstand = 0.5;
+        if (Math.abs(a - b) <= Math.abs(afstand)){
+            return true;
+        }
+        else {
+            return false;
+        }
+    }
+
+    /**
+     * @summary Creates and return a new array by the second index e.g.: A[[1,2],[3,4]] with idx=1 becomes [2,4]
+     * @param {Array} arr the array from which to create the sub-array
+     * @param {number} idx the index to use as second index
+     * @returns {Array} return an array
+     */
+    function arrayBySecondIndex(arr, idx){
+        let resArray = new Array();
+        for (let q = 0; q < arr.length; q++) {
+            resArray[q] = arr[q][idx - 1];
```

Comment:
Indecies skal starte med 0. Det der er der ingen der regner med

Reply:
I no understand, what is problem?

Reply:
??????????